src/csharp/Grpc.IntegrationTesting/SslCredentialsTest.cs

```diff
@@ -188,6 +189,52 @@ public void Constructor_NullRootCerts()
             Assert.Throws(typeof(ArgumentNullException), () => new SslServerCredentials(keyCertPairs, null, SslClientCertificateRequestType.RequestAndRequireAndVerify));
         }
 
+        [Test]
+        public async Task VerifyPeerCallback_Accepted()
+        {
+            string targetNameFromCallback = null;
+            string peerPemFromCallback = null;
+            InitClientAndServer(
+                clientAddKeyCertPair: false,
+                clientCertRequestType: SslClientCertificateRequestType.DontRequest,
+                verifyPeerCallback: (ctx) =>
+                {
+                    targetNameFromCallback = ctx.TargetName;
```

Comment:
nit: related to the documentation comment, suggest making this test's `verifyPeerCallback` implementation thread safe. For example, it could acquire a mutex and then set `targetNameFromCallback` and `peerPemFromCallback` only on the first invocation.

..... just in case people use this test as an example.